### PR TITLE
Keep order of config file entries when converting YAML to JSON or JSON to YAML

### PR DESCRIPTION
**Pull request by fhagemann**
To enhance the YAML support, I have added `DataStructures` to the dependencies.

This allows parsing the config files in an `OrderedDict`. Thus, when converting YAML to JSON with `yaml2json` or JSON to YAML with `json2yaml`, the order of the entries is kept.

The only case in which this converter fails is when trying to convert an "empty" YAML file consisting of only white spaces and new lines to JSON.

### CODE REVIEW

**Comment by oschulz** [comment]
Usually, JSON tools don't preserve order, but I guess it can't hurt, and the users will probably appreciate this.

**Comment by fhagemann** [comment]
This YAML file:
```yaml
name: "Simple True Coax"
units:
  length: "mm"
  angle: "deg"
grid:
  coordinates: "cylindrical"
  axes:
    r: 45
    z:
      from: -10
      to: 70.0
objects:
  - type: "semiconductor"
    material: "HPGe"
    geometry:
      type: "difference"
      parts:
        - type: "tube"
          r: 40
          h: 60
        - type: "tube"
          r: 5
          h: 60
```

This used to be converted in a "random" order to JSON, especially putting `"type"` of `"geometry"` at the end or putting the `"name"` of the detector somewhere in the middle. This can be quite messy when converting long config files.
```json
{
    "units": {
        "length": "mm",
        "angle": "deg"
    },
    "name": "Simple True Coax",
    "objects": [
        {
            "material": "HPGe",
            "geometry": {
                "parts": [
                    {
                        "r": 40,
                        "type": "tube",
                        "h": 60
                    },
                    {
                        "r": 5,
                        "type": "tube",
                        "h": 60
                    }
                ],
                "type": "difference"
            },
            "type": "semiconductor"
        }
    ],
    "grid": {
        "coordinates": "cylindrical",
        "axes": {
            "r": 45,
            "z": {
                "to": 70.0,
                "from": -10
            }
        }
    }
}

```




With this pull request it will be ordered:
```json
{
    "name": "Simple True Coax",
    "units": {
        "length": "mm",
        "angle": "deg"
    },
    "grid": {
        "coordinates": "cylindrical",
        "axes": {
            "r": 45,
            "z": {
                "from": -10,
                "to": 70.0
            }
        }
    },
    "objects": [
        {
            "type": "semiconductor",
            "material": "HPGe",
            "geometry": {
                "type": "difference",
                "parts": [
                    {
                        "type": "tube",
                        "r": 40,
                        "h": 60
                    },
                    {
                        "type": "tube",
                        "r": 5,
                        "h": 60
                    }
                ]
            }
        }
    ]
}

```

In addition, it is also possible to easily convert JSON to YAML files which allows adding comments, for example.

**Review by schustermartin — APPROVED** [review]
Looks good.

File: src/SolidStateDetector/DetectorGeometries.jl [hunk 1]
```diff
@@ -55,6 +55,22 @@ function yaml2json(directory::String)# or filename
...
+
+function json2yaml_convert(filename::String)
+    data = JSON.parsefile(filename, dicttype = DataStructures.OrderedDict)
+    YAML.write_file(replace(filename, ".json" => ".yaml"), data)
+end
+function json2yaml(directory::String)# or filename
+    if endswith(directory, ".json")
+        json2yaml_convert(directory)
+    else
+        yamlfiles = filter(x->endswith(x,".json"),readdir(directory))
```
Comment (fhagemann) [inline src/SolidStateDetector/DetectorGeometries.jl:67]: I made a small mistake here. It should be:
```julia
yamlfiles = filter(x->endswith(x,".json"), joinpath.(directory, readdir(directory)))
```
Same above in line 52.

And for consistency, we should rename `yamlfiles` in the `json2yaml` function to `jsonfiles`.

**Pull request merged**